shipment_advice_invoice_auto_complete/models/shipment_advice.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class ShipmentAdvice(models.Model):
+    _inherit = "shipment.advice"
+
+    supplier_ids = fields.Many2many(
+        comodel_name="res.partner",
+        compute="_compute_supplier_ids",
+        store=True,
+        copy=False,
+    )
+    is_invoiced = fields.Boolean(compute="_compute_is_invoiced", store=True, copy=False)
+    invoice_line_ids = fields.One2many(
+        "account.move.line",
+        "shipment_advice_id",
+        string="Bill Lines",
+        readonly=True,
+        copy=False,
+    )
+
+    @api.depends("planned_move_ids.purchase_line_id.order_id.partner_id")
+    def _compute_supplier_ids(self):
+        for shipment in self:
+            if shipment.shipment_type == "incoming":
+                shipment.supplier_ids = (
+                    shipment.planned_move_ids.purchase_line_id.order_id.partner_id
+                )
+            else:
+                shipment.supplier_ids = False
+
+    @api.depends("planned_move_ids", "invoice_line_ids")
+    def _compute_is_invoiced(self):
+        for shipment in self:
+            if shipment.state != "done" or shipment.shipment_type != "incoming":
+                shipment.is_invoiced = False
+                continue
+            # TODO: improve performance, maybe
+            moves = shipment.planned_move_ids.filtered(
+                lambda move: move.state == "done" and move.purchase_line_id
+            )
+            for move in moves:
+                invoice_lines = shipment.invoice_line_ids.filtered(
+                    lambda line: line.purchase_line_id == move.purchase_line_id
+                    and line.move_id.state != "cancel"
+                )
+                if not sum(invoice_lines.mapped("quantity")) == move.quantity_done:
+                    shipment.is_invoiced = False
+                    break
+            else:
+                shipment.is_invoiced = True
```

Comment:
We need more methods. :-)
In order to make this work also for kit products (products with phantom bom). 
A Kit Product will be split in multiple moves, therefor we have to calculate the quantity done different. 
To make this happen we need a separate method for quantity done. 

We have also to care about different uoms. 

Something like this. 
```suggestion
    
    def _get_quantity_done(self, moves):
        qty_done = 0
        for move in moves:
            if move.state != "done":
                continue
            qty_done += move.product_uom._compute_quantity(move.quantity_done, move.product_id.uom_id)
        return qty_done
    
    def _get_qties_invoiced_by_purchase_line(self):
        self.ensure_one()
        qties_invoiced = {}
        for invoice_line in sef.invoice_line_ids:
            if invoice_line.move_id.state == 'cancel' or not invoice_line.purchase_line_id:
                continue
            qties_invoiced.setdefault(invoice_line.purchase_line_id, 0)
            qties_invoiced[invoice_line.purchase_line_id] += invoice_line.product_uom_id._compute_quantity(invoice_line.quantity, invoice_lien.product_id.uom_id)
        return qties_invoiced
    
    def _is_invoiced(self):
        if self.state != "done" or self.shipment_type != "incoming":
            return False
        is_invoiced = True
        qties_invoiced_by_purchase_line = self._get_qties_invoiced_by_purchase_line()
        for purchase_line, moves in groupby(self.planned_move_ids, lambda m: move.purchase_line_id):
            if not purchase_line:
                continue
            invoice_qty = qties_invoiced_by_purchase_line[purchase_line]
            quantity_done = self._get_quantity_done(moves)
            if float_compare(invoiced_qty, quantity_done, move.product_id.uom_id.rounding) < 0:
                is_invoiced = False
                break
        return is_invoiced
    
    @api.depends("planned_move_ids", "invoice_line_ids")
    def _compute_is_invoiced(self):
        for shipment in self:
            shipment.is_invoiced = shipment._is_invoiced()
```